src/io/flutter/module/FlutterModuleBuilder.java

```diff
@@ -107,6 +109,21 @@ private static void createProjectFiles(@NotNull final ModifiableRootModel model,
     }
   }
 
+  @Override
+  public boolean validateModuleName(@NotNull String moduleName) throws ConfigurationException {
+    if (!FlutterUtils.isValidDartdentifier(moduleName)) {
+      throw new ConfigurationException("Invalid Flutter module name '" + moduleName + "'; must be a valid Dart identifier.");
```

Comment:
It's not obvious why the IDEA module name needs to be a valid Dart identifier. Add a comment explaining this?